lib/pkg.ml

```diff
@@ -89,7 +89,15 @@ let readme p =
 let opam p =
   match p.opam with
   | Some f -> Ok f
-  | None -> name p >>| fun name -> Fpath.v (name ^ ".opam")
+  | None -> (
+      name p >>= fun name ->
+      let filename = name |> Fpath.v |> Fpath.add_ext ".opam" in
+      let opam_subdir = Fpath.v "opam" in
+      let in_opam_subdir = Fpath.append opam_subdir filename in
+      match OS.File.exists in_opam_subdir with
```

Comment:
There's probably a `Rresult.R.map` or `Result.map` to use if you'd like to save a line